src/main/scala/beam/agentsim/agents/ridehail/repositioningmanager/DemandFollowingRepositioningManager.scala

```diff
@@ -130,17 +130,23 @@ class DemandFollowingRepositioningManager(val beamServices: BeamServices, val ri
   private def findWhereToReposition(tick: Int, vehicleLocation: Coord, vehicleId: Id[Vehicle]): Option[Coord] = {
     val currentHour = tick / 3600
     val nextHour = currentHour + 1
+    val percentageOfClosestClusters =
+      beamServices.beamConfig.beam.agentsim.agents.rideHail.repositioningManager.demandFollowingRepositioningManager.PercentageOfClosestClustersToConsider
+
     hourToClusters.lift(nextHour).map { clusters =>
-      // We get top 5 closest clusters and randomly pick one of them.
+      val N = (clusters.length * percentageOfClosestClusters).toInt
+
+      // We get top N closest clusters and randomly pick one of them.
       // The probability is proportional to the cluster size - meaning it is proportional to the demand, as higher demands as higher probability
-      val top5Closest = clusters.sortBy(x => beamServices.geo.distUTMInMeters(x.coord, vehicleLocation)).take(5)
-      val pmf = top5Closest.map { x =>
+      val topNClosest = clusters.sortBy(x => beamServices.geo.distUTMInMeters(x.coord, vehicleLocation)).take(N)
+      val pmf = topNClosest.map { x =>
         new CPair[ClusterInfo, java.lang.Double](x, x.size.toDouble)
       }.toList
+
       val distr = new EnumeratedDistribution[ClusterInfo](rng, pmf.asJava)
       val sampled = distr.sample()
       logger.debug(
-        s"tick $tick, currentHour: $currentHour, nextHour: $nextHour, vehicleId: $vehicleId, vehicleLocation: $vehicleLocation. Top 5 closest: ${top5Closest.toVector}, sampled: $sampled"
+        s"tick $tick, currentHour: $currentHour, nextHour: $nextHour, vehicleId: $vehicleId, vehicleLocation: $vehicleLocation. Top 5 closest: ${topNClosest.toVector}, sampled: $sampled"
```

Comment:
Change "Top 5" in this string to the actual number being used, e.g.: ("Top {}",N)